### PR TITLE
Search history: Design review

### PR DESCRIPTION
**Pull request by ashiagr**
Part of #752

## Description

This includes search history design improvements to match the specs: 8JZyr1ZkWiGn3kSjJYvpoZ-fi-160%3A5856
APK for design review attached in pdeCcb-1iH-p2#comment-1675.

## Screenshots or Screencast 

Light Theme | Different Theme | Large font
---|----|---
 | | 

|Landscape mode|
|----|
||

## Checklist
- If this is a user-facing change, I have added an entry in CHANGELOG.md
- I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [x] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [x] for accessibility with TalkBack

### CODE REVIEW

**Comment by CookieyedCodes** [comment]
Would it be a nice improvement that if a searched podcast is not subscribed that their is a plus icon to subscribe to it on from the history screen(episodes & folders would have a go to folder & go to episode icon representing by an arrow instead > 🤔)

**Comment by ashiagr** [comment]
> Would it be a nice improvement that if a searched podcast is not subscribed that their is a plus icon to subscribe to it on from the history screen(episodes & folders would have a go to folder & go to episode icon representing by an arrow instead > 🤔)

We're eliminating subscribe icon from search history rows as it felt too cluttered to have it along with the close button and to keep search history rows lightweight. It will be present in the search result rows. With that, there's no need to show plus or arrow icon.

> Also maby have a setting for a condensed view the padding feels a tad excessive 🤔🤔

Nice idea :) thanks for sharing!

**Comment by CookieyedCodes** [comment]
That makes more sense 😅, pinning if anything would still be nice at that tho 🤔 (maby on a long press if you want to cut down on icons 🤔🤔)

**Comment by ashiagr** [comment]
Addressed feedback: pdeCcb-1iH-p2#comment-1693 
cc @adamzelinski 

> Can we please update the search field empty string to say ‘Search podcasts or add URL’

> Can we please update the delete button to read ‘Clear Search History’ I think delete seems a bit harsh.

Search field empty text | Clear Search History 
---|---
 |

**Comment by CookieyedCodes** [comment]
I'd personally say add RSS URL that would give better clarity in my view 🤔🤔

**Review by mchowning — APPROVED** [review]
This looks great. Nice job! 🎉 

The only minor issue I noticed was that when tapping on a search term in the history, it looks like we still have the couple of seconds delay before the remote search is initiated (like if you had typed the search term). I wonder if it might be better to instantly do start the remote search in that scenario. It's most noticeable when searching from the Discover screen. 

Certainly not a blocker for merging this PR, just something to think about going forward.

<details>
<summary>Video</summary>

https://user-images.githubusercontent.com/4656348/219454841-7e0eb0aa-3976-4467-a6ba-bf09ab909d1c.mov


</details>

File: modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/db/dao/SearchHistoryDao.kt [hunk 1]
```diff
@@ -25,4 +24,7 @@ abstract class SearchHistoryDao {
 
     @Query("DELETE FROM search_history")
     abstract suspend fun deleteAll()
+
+    @Query("DELETE FROM search_history where _id NOT IN (SELECT _id from search_history ORDER BY modified DESC LIMIT :limit)")
```
[the remark this answers is not in the document]
Comment (ashiagr) [reply modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/db/dao/SearchHistoryDao.kt:28]: Thank you for adding the test! It is definitely useful. 🙏

**Comment by ashiagr** [comment]
> The only minor issue I noticed was that when tapping on a search term in the history, it looks like we still have the couple of seconds delay before the remote search is initiated (like if you had typed the search term). I wonder if it might be better to instantly do start the remote search in that scenario. It's most noticeable when searching from the Discover screen.

Good point. I fixed it in 4227b91.

**Comment by adamzelinski** [comment]
> I'd personally say add RSS URL that would give better clarity in my view

You're right, it might be better to use add RSS URL. Can we please make this change?

**Comment by ashiagr** [comment]
> I'd personally say add RSS URL that would give better clarity in my view

> You're right, it might be better to use add RSS URL. Can we please make this change?

Changed in 29956f9.

Thanks everyone for great suggestions 🙏

**Comment by CookieyedCodes** [comment]
Well it looks good here 


But it seems folders share the same text now

**Comment by ashiagr** [comment]
Raised beta fix: https://github.com/Automattic/pocket-casts-android/pull/803. Thanks for pointing it out, @CookieyedCodes 🙏

**Pull request merged**